### ISSUE TITLE
MongoDB: Using `delete` on a one-to-one relation while updating removes other relations

### ISSUE THREAD

**Issue by mannol** [prisma/prisma#17255]
### Bug description

It seems that, if you try to delete a relation while doing the update, all the other relations get deleted too

### How to reproduce

See the code bellow. But basically, if you do this (try to delete alice):

```TS
await prisma.main.update({
    where: { id: mainId.toString() },
    data: { alice: { delete: true } }, 
});
```

unrelated ONE-TO-ONE relation entry (of model BOB) gets deleted for some reason.

But, if you do this:

```TS
await prisma.main.update({
    where: { id: mainId.toString() },
    data: { alice: { disconnect: true } },
});
```

It's fine. 

## Logs
These are logs generated while executing the update operation:

```JS
db.Main.aggregate([ { $match: { $expr: { $and: [ { $and: [ { $and: [ { $eq: [ "$_id", { $literal: ObjectId("63bf216e94bcc9074a67460d"), }, ], }, { $ne: [ "$_id", "$$REMOVE", ], }, ], }, ], }, { }, ], }, }, }, { $project: { _id: 1, }, }, ])
db.Main.aggregate([ { $match: { $expr: { $or: [ { $and: [ { $and: [ { $eq: [ "$_id", { $literal: ObjectId("63bf216e94bcc9074a67460d"), }, ], }, { $ne: [ "$_id", "$$REMOVE", ], }, ], }, ], }, ], }, }, }, { $project: { _id: 1, aliceId: 1, }, }, ])
db.Alice.aggregate([ { $match: { $expr: { $and: [ { }, { $and: [ { $in: [ "$_id", [ { $literal: ObjectId("63bf216e94bcc9074a67460f"), }, ], ], }, { $ne: [ "$_id", "$$REMOVE", ], }, ], }, ], }, }, }, { $project: { _id: 1, }, }, ])
db.Main.aggregate([ { $match: { $expr: { $and: [ { }, { $and: [ { $in: [ "$aliceId", [ { $literal: ObjectId("63bf216e94bcc9074a67460f"), }, ], ], }, { $ne: [ "$aliceId", "$$REMOVE", ], }, ], }, ], }, }, }, { $project: { _id: 1, aliceId: 1, }, }, ])
db.Bob.aggregate([ { $match: { $expr: { $and: [ { }, { $and: [ { $in: [ "$mainId", [ { $literal: ObjectId("63bf216e94bcc9074a67460d"), }, ], ], }, { $ne: [ "$mainId", "$$REMOVE", ], }, ], }, ], }, }, }, { $project: { _id: 1, mainId: 1, }, }, ])
db.Bob.deleteMany({ _id: { $in: [ ObjectId("63bf216e94bcc9074a67460e"), ], }, })
db.Main.updateMany({ _id: { $in: [ ObjectId("63bf216e94bcc9074a67460d"), ], }, },{ $set: { aliceId: { $literal: null, }, }, })
db.Alice.deleteMany({ _id: { $in: [ ObjectId("63bf216e94bcc9074a67460f"), ], }, })
db.Main.aggregate([ { $match: { $expr: { $and: [ { $and: [ { $eq: [ "$_id", { $literal: ObjectId("63bf216e94bcc9074a67460d"), }, ], }, { $ne: [ "$_id", "$$REMOVE", ], }, ], }, ], }, }, }, { $project: { _id: 1, aliceId: 1, }, }, ])
```

### Expected behavior

In the example above, we never wanted for Bob to be deleted

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

model Main {
  id String @id @map("_id") @db.ObjectId

  alice   Alice?  @relation(fields: [aliceId], references: [id], onDelete: SetNull, onUpdate: Cascade)
  aliceId String? @db.ObjectId

  bob Bob?
}

model Alice {
  id        String @id @map("_id") @db.ObjectId
  manyMains Main[]
}

model Bob {
  id String @id @map("_id") @db.ObjectId

  main   Main   @relation(fields: [mainId], references: [id], onDelete: Cascade, onUpdate: Cascade)
  mainId String @unique @db.ObjectId
}
```

```ts
import { Prisma, PrismaClient } from "@prisma/client";
import { ObjectID } from "bson";

const prisma = new PrismaClient();

async function main() {
  const mainId = new ObjectID();
  const bobId = new ObjectID();
  const aliceId = new ObjectID();

  await prisma.main.create({
    data: {
      id: mainId.toString(),
      bob: {
        create: {
          id: bobId.toString(),
        },
      },
      alice: {
        create: {
          id: aliceId.toString(),
        },
      },
    },
  });

  const bobCountBefore = await prisma.bob.count();
  console.log("BOB count before:", bobCountBefore);

  // now, update the main instance and delete alice
  await prisma.main.update({
    where: { id: mainId.toString() },
    data: { alice: { delete: true } },
  });

  // the above will also cause bob to be deleted, wow!!
  const bobCountAfter = await prisma.bob.count();
  console.log("BOB count after:", bobCountAfter);
}

main()
  .then(async () => {
    await prisma.$disconnect();
  })
  .catch(async (e) => {
    console.error(e);
    await prisma.$disconnect();
    process.exit(1);
  });
```

### Environment & setup

- OS: macOS, MongoDB runs on Linux Docker Container
- Database: MongoDB
- Node.js version: v16.14.2

### Prisma Version

```
prisma                  : 4.8.1
@prisma/client          : 4.8.1
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine d6e67a83f971b175a593ccc12e15c4a757f93ffe (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Migration Engine        : migration-engine-cli d6e67a83f971b175a593ccc12e15c4a757f93ffe (at node_modules/@prisma/engines/migration-engine-darwin-arm64)
Introspection Engine    : introspection-core d6e67a83f971b175a593ccc12e15c4a757f93ffe (at node_modules/@prisma/engines/introspection-engine-darwin-arm64)
Format Binary           : prisma-fmt d6e67a83f971b175a593ccc12e15c4a757f93ffe (at node_modules/@prisma/engines/prisma-fmt-darwin-arm64)
Format Wasm             : @prisma/prisma-fmt-wasm 4.8.0-61.d6e67a83f971b175a593ccc12e15c4a757f93ffe
Default Engines Hash    : d6e67a83f971b175a593ccc12e15c4a757f93ffe
Studio                  : 0.479.0
```

**Comment by mannol** [comment]
Also, it appears that, when this happens, Prisma deletes *ALL* relations, including one-to-many and many-to-many...

**Comment by janpio** [comment]
Thanks for providing the perfect reproduction, we are looking into this right now.

**Comment by Weakky** [comment]
Hey @mannol, thanks for reporting this bug! It will be included in Prisma's 4.9.0 release.
